modules/phase_field/src/ics/SmoothCircleBaseIC.C

```diff
@@ -134,6 +138,5 @@ SmoothCircleBaseIC::computeCircleGradient(const Point & p, const Point & center,
   if (dist != 0.0)
     return _mesh.minPeriodicVector(_var.number(), center, p) * (DvalueDr / dist);
   else
-    return Gradient(0.0, 0.0, 0.0);
-
+    return 0.0;
```

Comment:
I guess you are saving the cost of an object construction, maybe it's not so bad.